api/preprints/views.py

```diff
@@ -69,6 +69,30 @@
 from osf.metrics import PreprintDownload, PreprintView
 
 
+class RejectPreprintEditMixin:
+    '''Override method to reject modify requests for old preprint versions (except for withdrawal)'''
+    def update(self, request, *args, **kwargs):
+        preprint = self.get_preprint(check_object_permissions=False)
+        if preprint.is_latest_version:
+            return super().update(request, *args, **kwargs)
+
+        raise Conflict(detail='Edit availiabe only for the last preprint version or withdrawn preprints')
```

Comment:
I think `405 Method Not Allowed` is the best fit for all three errors. I don't think we have one defined in the code base, you probably want to define it next to where `Conflict` is.

Reply:
I think that `409 Conflict` is better suited for this case, because in theory this request can be executed, but for a correct version of the preprint

`405 Method Not Allowed` is more like the method is not allowed at all (for example, if we try to send a POST request to an endpoint where only GET is allowed)